src/plugins/explore/public/components/visualizations/style_panel/threshold_custom_values.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFieldNumber,
+  EuiButtonIcon,
+  EuiButton,
+  EuiSpacer,
+  EuiColorPicker,
+} from '@elastic/eui';
+import { ThresholdRangeValue } from '../types';
+import { useDebouncedValue, useDebouncedNumericValue } from '../utils/use_debounced_value';
+
+export interface RangeProps {
+  index: number;
```

Comment:
Suggest to rename it to `id`:
```suggestion
  id: number;
```